src/DefaultState.ts

```diff
@@ -0,0 +1,57 @@
+import { MouseEvent } from 'react';
+import {
+  SelectingState,
+  State,
+  Action,
+  InputType,
+  ActionEvent,
+  DragCanvasState,
+} from '@projectstorm/react-canvas-core';
+import {
+  PortModel,
+  DiagramEngine,
+  DragDiagramItemsState,
+} from '@projectstorm/react-diagrams-core';
+
+export class DefaultState extends State<DiagramEngine> {
+  dragCanvas: DragCanvasState;
+  dragItems: DragDiagramItemsState;
+  isSelection: boolean;
+
+  constructor() {
+    super({ name: 'starting-state' });
+    this.childStates = [new SelectingState()];
+    this.dragCanvas = new DragCanvasState();
+    this.isSelection = true;
+    this.dragItems = new DragDiagramItemsState();
+
+    // determine what was clicked on
+    this.registerAction(
+      new Action({
+        type: InputType.MOUSE_DOWN,
+        fire: (event: ActionEvent<MouseEvent | any>) => {
+          const element = this.engine
+            .getActionEventBus()
+            .getModelForEvent(event);
+
+          // the canvas was clicked on, transition to the dragging canvas state
+          if (!element) {
+            this.transitionWithEvent(this.dragCanvas, event);
+          }
+          // initiate dragging a new link
+          else if (element instanceof PortModel) {
+            return;
```

Comment:
Just to confirm, the dragging of new links is something we will need to take care of later on correct? If that's true maybe add a todo in this part of the code to make that clear.